openssl/src/ossl_param.rs

```diff
@@ -37,6 +39,78 @@ foreign_type_and_impl_send_sync! {
     pub struct OsslParamArrayRef;
 }
 
+impl OsslParamArray {
+    /// Locates the individual `OSSL_PARAM` element identified by the key
+    /// in the `OsslParamArray` array and returns a reference to it.
+    #[corresponds(OSSL_PARAM_locate)]
+    #[allow(dead_code)] // TODO: remove when when used by ML-DSA / ML-KEM
+    pub(crate) fn locate<'a>(&'a self, key: &CStr) -> Result<OsslParam<'a>, ErrorStack> {
+        unsafe {
+            let param = cvt_p(ffi::OSSL_PARAM_locate(self.as_ptr(), key.as_ptr()))?;
+            Ok(OsslParam::from_ptr(param))
+        }
+    }
+}
+
+/// A reference to a single `OSSL_PARAM` structure inside
+/// an `OsslParamArray`.
+pub(crate) struct OsslParam<'a>(&'a ffi::OSSL_PARAM);
+
+impl<'a> OsslParam<'a> {
+    /// Constructs a reference to a single `OSSL_PARAM` structure
+    /// from a pointer to an interior element of an OSSL_PARAM array.
+    ///
+    /// # Safety
+    ///
+    /// The pointer must be valid and point to an `OSSL_PARAM` structure.
+    unsafe fn from_ptr(ptr: *const ffi::OSSL_PARAM) -> OsslParam<'a> {
+        OsslParam(&*ptr)
+    }
+
+    /// Returns a pointer to the underlying `OSSL_PARAM` structure.
+    unsafe fn as_ptr(&self) -> *const ffi::OSSL_PARAM {
+        self.0 as *const ffi::OSSL_PARAM
+    }
+
+    /// Get `BigNum` from this underlying OSSL_PARAM.
+    #[corresponds(OSSL_PARAM_get_BN)]
+    #[allow(dead_code)] // TODO: remove when when used by ML-DSA / ML-KEM
+    pub(crate) fn get_bn(&self) -> Result<&'a BigNum, ErrorStack> {
+        unsafe {
+            let mut bn: *mut ffi::BIGNUM = ptr::null_mut();
+            cvt(ffi::OSSL_PARAM_get_BN(self.as_ptr(), &mut bn))?;
+            Ok(&*(bn as *const BigNum))
```

Comment:
This should use `from_ptr`

Reply:
I think I had `from_ptr` but switched away so I could annotate with a lifetime.

But I think we don't need this method at all. Let me see if I can simplify this.